File: colt-steele/isSubsequence.js
// isSubsequence takes in two strings and checks whether the characters in the first string form a subsequence of the characters in the second string. IN other words, the fn should check whether the characters in the first string appear somewhere in the second string WITHOUT THEIR ORDER CHANGING 

// Attempt 1 (correct)
function isSubsequence(str1, str2) {
    let i = 0;
    let j = 0;
      while (j < str2.length && i < str1.length) {
        if (str1[i] == str2[j]) {
          i++;
          j++;
        } else if (str1[i] !== str2[j]) {
          j++;
        }
      }
      if (i === str1.length) {
          return true;
      } else {
          return false;
      }
}

// Their solution 
function isSubsequence(str1, str2) {
  var i = 0;
  var j = 0;
  if (!str1) return true;
  while (j < str2.length) {
    if (str2[j] === str1[i]) i++;
    if (i === str1.length) return true;
    j++;
  }
  return false;
}

File: colt-steele/Basics/isSubsequence.rb
def is_subsequence(str1, str2)
    i = 0
    j = 0
    if (!str1) 
        return true
    end
    while (j < str2.length) do
        if str2[j] === str1[i]
            i += 1
        end
        if i === str1.length
            return true
        end
        j += 1
    end
    false
end

puts is_subsequence('abc', 'acadabra')